File: app/schemas/social_account.py
from typing import Optional, List, Dict, Any
from pydantic import BaseModel
from datetime import datetime


class SocialAccountBase(BaseModel):
    uid: str
    name: str
    reaction_count: Optional[int] = 0
    phone_number: Optional[str] = None
    status_id: Optional[int] = None
    type_id: Optional[int] = None
    note: Optional[str] = None
    is_active: Optional[bool] = True


class SocialAccountCreate(SocialAccountBase):
    pass


class SocialAccountUpdate(SocialAccountBase):
    uid: Optional[str] = None
    name: Optional[str] = None


class SocialAccount(SocialAccountBase):
    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class SocialAccountInDB(SocialAccount):
    """Schema for social account stored in database"""
    pass


class SocialAccountWithRelations(SocialAccount):
    unit: Optional[Dict[str, Any]] = None
    task: Optional[Dict[str, Any]] = None
    status_name: Optional[str] = None

    class Config:
        from_attributes = True 

File: front-end/src/app/pages/SocialAccount/SocialAccount.ts
export interface SocialAccountModal {
    uid: string,
    name: string,
    phone_number?: string,
    status_id: number,
    type_id: number,
    note: string,
    reaction_number: number,
    characteristics_id: number,
    unit_id: string,
    task_id: number,
    is_active: Boolean,
}
export interface SocialAccount extends SocialAccountModal {
    trangthai_name: string,
    ctnv_id: number,
    donvi_id: string
}
export interface SocialAccountResponse {
    id: number
    uid: string
    name: string
    reaction_count: number
    phone_number: string
    status_id: number
    status_name?: string
    type_id: number
    note: string
    is_active: boolean
    created_at: string
    updated_at: string
    unit?: {
        id: string
        name: string
    }
    task?: {
        id: number
        name: string
    }
}

export interface SocialAccountSimple {
    uid: string
    name: string
}

export interface SocialAccountTypeGroup {
    type_id: number
    type_name: string
    data: SocialAccountSimple[]
}

export interface SocialAccountTypeResponse {
    data: SocialAccountTypeGroup[]
}

export interface SocialAccountListResponse {
    account_type_name: string
    data: SocialAccountResponse[]
}
export interface status {
    created_at: string,
    id: number,
    name: string,
    color: string,
    updated_at: string
}
export interface unit {
    created_at:string,
    id: string,
    name:string,
    updated_at:string
}
export interface account_type extends unit{

}
export interface task extends unit{

}
export interface relationship extends unit{

}
export interface characteristics extends unit{

}
